client/packages/common/src/ui/layout/tables/components/Cells/MultiNumberInputCell/MultiNumberInputCell.tsx

```diff
@@ -0,0 +1,167 @@
+  const [buffer, setBuffer] = useBufferState(column.accessor({ rowData }));
+  const updater = useDebounceCallback(column.setter, [column.setter], 350);
+
+  const autoFocus = rowIndex === 0 && columnIndex === 0;
+
+  const cellValues = getCellValues(buffer as number | undefined, units);
+
+  return (
+    <Box display="flex">
+      {units.map((unit, index) => (
+        <NumericTextInput
+          key={unit.key}
+          id={id}
+          disabled={isDisabled}
+          autoFocus={autoFocus}
+          {...TextInputProps}
+          InputProps={{
+            sx: { '& .MuiInput-input': { textAlign: 'right' } },
+            ...TextInputProps?.InputProps,
+          }}
+          onChange={num => {
+            const newValue = num === undefined ? min : num;
+            if (cellValues[index] === newValue) return;
+            const newValues = [...cellValues];
+            newValues[index] = newValue;
+            const newTotal = computeTotal(newValues, units);
+            setBuffer(newTotal);
+            updater({ ...rowData, [column.key]: Number(newTotal) });
+          }}
+          min={min}
+          // Limit smaller units to less than the value of the next highest
+          // unit, e.g. "Months" would be limited to 11.9999
+          max={setMax(units, index, max)}
+          decimalLimit={decimalLimit}
+          step={step}
+          multiplier={multiplier}
+          allowNegative={allowNegative}
+          defaultValue={defaultValue}
+          value={cellValues[index]}
+          width={width}
+          endAdornment={unit.label}
+        />
+      ))}
+    </Box>
+  );
+};
+
+export const getCellValues = (value: number | undefined, units: Unit[]) => {
+  if (value === undefined) return units.map(_ => 0);
+
+  let remainder = value;
+  return units.map((unit, index) => {
+    const thisValue =
+      index === units.length - 1
+        ? remainder
+        : Math.floor(remainder / unit.ratio);
+    // Rounding required to binary precision problems
+    // e.g. 66.6 % 12 => 6.599999999999994
+    remainder = parseFloat((remainder % unit.ratio).toFixed(5));
+    return thisValue;
+  });
+};
+
+export const computeTotal = (values: number[], units: Unit[]) =>
+  values.reduce((sum, val, index) => {
+    const ratio = units[index]?.ratio ?? 1;
+    return sum + ratio * val;
+  }, 0);
+
+export const setMax = (
+  units: Unit[],
+  index: number,
+  max: number | undefined
+) => {
+  if (index === 0) return max;
+  const thisRatio = (units[index] as Unit).ratio;
+  const higherRatio = (units[index - 1] as Unit).ratio;
+  // If not the last value, should be limited to the integer below the next
+  // highest ratio. But for the last value, we need to be able to accept
+  // fractional input right up to the ratio, so we allow this to an arbitrary
+  // decimal precision (can't see needing more than 4 d.p for this kind of
+  // thing!)
+  const limitReduction = index === units.length - 1 ? 0.0001 : 1;
```

Comment:
Given how narrow the cells are, I think might need to go with 1 or 2 dp (maybe needs to be configurable for different use cases, but that can be done in future as needed) ... I thought this was jumping to `999`


https://github.com/user-attachments/assets/b433fc68-37c4-41a8-816a-bbc236bc838d

Reply:
Yes, for sure, there's no reason to ever need decimals for anything other than the smallest unit.

Reply:
I actually thought there was a default limit of 2dp, but it doesn't seem to be the case here. I've added that to this component now.

Reply:
Having said that, the limit here is just for internal precision, so the num of dp here just needs to be more than anything we'd realistically expect to be used, hence 4 seemed reasonable.